plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultQueryBuilder.java

```diff
@@ -239,20 +240,29 @@ protected String getRelation(JdbcClient client, RemoteTableName remoteTableName)
         return client.quoted(remoteTableName);
     }
 
-    protected String getProjection(JdbcClient client, List<JdbcColumnHandle> columns, Map<String, String> columnExpressions)
+    protected String getProjection(JdbcClient client, List<JdbcColumnHandle> columns, Map<String, String> columnExpressions, Optional<List<List<JdbcColumnHandle>>> groupingSets)
     {
-        if (columns.isEmpty()) {
-            return "1 x";
+        boolean implicitGlobalAggregation = groupingSets.isPresent() && groupingSets.get().equals(ImmutableList.of(ImmutableList.of()));
+        List<String> additionalProjections = new ArrayList<>();
+        if (implicitGlobalAggregation) {
+            additionalProjections.add("count(*) x");
```

Comment:
This works (cc @Yao-MR), but it's also kind of pointless.
I will try do better.